appveyor_build.sh

```diff
@@ -41,33 +41,34 @@ function set_configuration {
     echo "Edit $FILE to set PREFIX=$2"
     sed -e "/PREFIX=/s|=.*|=$2|" \
         -e "/^ *CFLAGS *=/s/\r\?$/ $3\0/" \
-         config/Makefile.$1 > config/Makefile
+         "config/Makefile.$1" > config/Makefile
 #    run "Content of $FILE" cat config/Makefile
 }
 
-APPVEYOR_BUILD_FOLDER=$(echo $APPVEYOR_BUILD_FOLDER| cygpath -f -)
+APPVEYOR_BUILD_FOLDER=$(echo "$APPVEYOR_BUILD_FOLDER" | cygpath -f -)
 # These directory names are specified here, because getting UTF-8 correctly
 # through appveyor.yml -> Command Script -> Bash is quite painful...
-OCAMLROOT=$(echo $PROGRAMFILES/Бактріан🐫| cygpath -f - -m)
+OCAMLROOT=$(echo "$PROGRAMFILES/Бактріан🐫" | cygpath -f - -m)
 
 # This must be kept in sync with appveyor_build.cmd
 BUILD_PREFIX=🐫реализация
 
-export PATH=$(echo $OCAMLROOT| cygpath -f -)/bin/flexdll:$PATH
+PATH=$(echo "$OCAMLROOT" | cygpath -f -)/bin/flexdll:$PATH
+export PATH
```

Comment:
What is the purpose in splitting the `export` and assignment?

Reply:
See [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155).

Reply:
I sort of end up repeating the question, I'm afraid!

Reply:
IIUC, `export PATH=$(...)` will continue if `cygpath` explodes, despite `set -e`, while separate assignment and export will quit.

Reply:
Indeed, that's my understanding - but if cygpath explodes, there will be serious trouble shortly, including but probably not limited to the lack of flexlink in PATH. These failures will be obvious - accidentally moving the `PATH=` elsewhere and forgetting to move the `export` line with it on the other hand is a bit more subtle to figure out what you did, amidst long waits in the AppVeyor build queue!

Reply:
Hmm, thinking about this some more I don't think the `export` line is needed at all. `PATH` is always exported, so I think ordinary assignments will show up in the environment.

Reply:
Did the lint tool report a warning on this line? If not, I'm definitely siding with @dra27 here.

@stedolan you're right, unless somebody is foolish enough to do `export -n PATH`...